espoon-voltti/evaka
# [EVAKA-4039] enduser: fix decisions not loading on direct page navigation

#### Summary
- Regressed in 18e4ac451ccc9cd417e840c4614d5110ceb42d04
- There was an implicit dependency on applications being loaded into rootState when fetching decision summaries from state, and now that everything isn't loaded on authentication applications obviously weren't in the rootState
    - The dependency isn't actually necessary as all of the info is available directly in the decisions (childName was missing from the service response but available in already queried data)
- This fixes the issue where (because of the dependency) no decisions were listed when the user directly navigates to the decisions page from home without going through the applications page
- This also fixes nothing being loaded when the user directly navigates to a specific decision view instead of going through the list
    - Fix by triggering decisions loading if they aren't already loaded (no need to load twice if the user entered the detail view through the list view)
- Add new field "childName" to enduser decisions response type

Testing:

1. In the `master` version, login as an enduser with decisions (on the home page)
1. Navigate directly to the decisions page
1. -> children are loaded but no decisions are visible
1. Navigate directly to a decision's page (`/decisions/<uuid>`)
1. -> spinner is stuck on page, nothing is loaded



1. In this version, navigate the same way
1. -> decisions are loaded and shown
1. Navigate directly on a decision's page
1. -> decision details are loaded



1. In this version, navigate to the decisions list
1. In dev tools, see single request to `/decisions`
1. Open a decisions
1. In dev tools, see no new `/decisions` requests
